Libraries/Image/RCTImageLoader.m

```diff
@@ -480,6 +470,18 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
   __block void(^cancelLoad)(void) = nil;
   __weak RCTImageLoader *weakSelf = self;
 
+  // Load handlers are executed before the generic caching logic so they are responsible
+  // for caching the images if necessary.
+  id<RCTImageURLLoader> loadHandler = [self imageURLLoaderForURL:imageURLRequest.URL];
```

Comment:
Let's add 2 optional methods to the URLLoader protocol:
1) shouldCacheLoadedImages (i.e. does the loader have its own image cache, defaults to YES)
2) requiresScheduling (i.e. do we need to manage the invocation of this loader on our URL queue, defaults to YES)